### PR TITLE
add poseidon hash interface

### PR DESCRIPTION
**Pull request by lightning-li**

### CODE REVIEW

File: ecc/bn254/fr/poseidon/poseidon.go [hunk 1]
```diff
@@ -104,10 +111,74 @@ func Poseidon(input ...*fr.Element) *fr.Element {
...
+
+func PoseidonBytes(input ...[]byte) []byte {
+	inputElements := make([]*fr.Element, len(input))
+	for i, ele := range input {
+		num := new(big.Int).SetBytes(ele)
+		if num.Cmp(fr.Modulus()) >= 0 {
+			panic("not support bytes bigger than modulus")
+		}
+		e := fr.Element{0, 0, 0, 0}
+		e.SetBigInt(new(big.Int).SetBytes(ele))
```
Comment (HSG88) [inline ecc/bn254/fr/poseidon/poseidon.go]: `e.SetBigInt(num)`

File: ecc/bn254/fr/poseidon/poseidon.go [hunk 2]
```diff
@@ -1,11 +1,18 @@
...
+	"errors"
 	"github.com/consensys/gnark-crypto/ecc/bn254/fr/poseidon/constants"
+	"hash"
+	"math/big"
 
 	"github.com/consensys/gnark-crypto/ecc/bn254/fr"
 )
 
+const (
+	BlockSize = fr.Bytes // BlockSize size that mimc consumes
```
Comment (HSG88) [inline ecc/bn254/fr/poseidon/poseidon.go]: `BlockSize = fr.Bytes // BlockSize size that Poseidon consumes`

**Review by HSG88 — APPROVED** [review]
Looks good to me

File: ecc/bn254/fr/poseidon/poseidon.go (hunk 1 above)
Comment (lightning-li) [reply ecc/bn254/fr/poseidon/poseidon.go]: Yeah. I fixed this

File: ecc/bn254/fr/poseidon/poseidon.go (hunk 2 above)
Comment (lightning-li) [reply ecc/bn254/fr/poseidon/poseidon.go]: Fixed

**Pull request merged**